python_scripts/01_tabular_data_exploration.py

```diff
@@ -188,20 +197,20 @@
 # These hand-written rules could work reasonably well without the need for any
 # machine learning. Note however that it is not very easy to create rules for
 # the region `40 < hours-per-week < 60` and `30 < age < 70`. We can hope that
-# machine learning can help in this region. Also note that visualization can
-# help creating hand-written rules but is limited to 2 dimensions (maybe 3
-# dimensions), whereas machine learning models can build models in
+# machine learning can help in this region. If visualization can
+# help creating hand-written rules, it is limited to 2 or 3 dimensions, 
+# when machine learning models can learn rules in
```

Comment:
Is "when" more clear than "whereas"?

The grammatical structure does not look very familiar to me but this is probably more my English skills that are problematic here ... I feel it is a bit too easy to parse it without the comma as "it is limited to 2 or 3 dimensions when machine learning can learn rules in high-dimensional space"